indexing-service/src/main/java/org/apache/druid/indexing/input/DruidInputSource.java

```diff
@@ -474,18 +541,40 @@ private static SortedMap<WindowedSegmentId, Long> createWindowedSegmentIdFromTim
   }
 
   public static List<TimelineObjectHolder<String, DataSegment>> getTimelineForInterval(
+      TaskToolbox toolbox,
       CoordinatorClient coordinatorClient,
       String dataSource,
       Interval interval
   )
   {
-    final Collection<DataSegment> usedSegments = FutureUtils.getUnchecked(
-        coordinatorClient.fetchUsedSegments(
-            dataSource,
-            Collections.singletonList(Preconditions.checkNotNull(interval, "interval"))
-        ),
-        true
-    );
+    Preconditions.checkNotNull(interval);
+
+    final Collection<DataSegment> usedSegments;
+    if (toolbox == null) {
+      usedSegments = FutureUtils.getUnchecked(
+          coordinatorClient.fetchUsedSegments(
+              dataSource,
+              Collections.singletonList(Preconditions.checkNotNull(interval, "interval"))
```

Comment:
We need not perform `checkNotNull` again. It has already been done at the start of the method.

Reply:
Ah, the check was already there